src/wp-includes/functions.wp-scripts.php

```diff
@@ -420,20 +420,32 @@ function wp_script_is( $handle, $status = 'enqueued' ) {
 }
 
 /**
- * Adds metadata to a script.
+ * Adds metadata to a registered script.
```

Comment:
Yes, this is true, as `WP_Scripts::add_data()` returns `false` when the script's handle is not registered.